### PR TITLE
Add examples for creating download links

### LINKED ISSUE #17734: 🐛 Bug Report: Link to PDF doesn't throws 404 error
### 📜 Description

I have a reference to PDF in my docs/onboarding.md file as below
**For reference:** 
```
This is a basic example of documentation. [here](./assets/onboarding.pdf)
```

When I click on it I am getting 404 error.
Is there any way to render the content of the PDF on the page

### 👍 Expected behavior

PDF should open on Techdocs page.

### 👎 Actual Behavior with Screenshots

It throws 404 Not Found

### 👟 Reproduction steps

1. Upload a pdf file under docs/assets/<file-name.pdf>
2. Use the PDF file reference in any .md file  as below
  ```
  This is a basic example of documentation. [here](./assets/onboarding.pdf)
  ```
3. Import the component in Backstage, visit component Techdocs and click on "here" hyperlink.
4. It will give you 404 error

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

node: v16.15.0
yarn: 1.22.19
cli:  0.22.5 (installed)
backstage:  1.12.1
```
Dependencies:
  @backstage/app-defaults                          1.2.1
  @backstage/backend-app-api                       0.4.1 
  @backstage/backend-common                        0.18.3
  @backstage/backend-dev-utils                     0.1.1 
  @backstage/backend-plugin-api                    0.5.0 
  @backstage/backend-tasks                         0.5.0 
  @backstage/catalog-client                        1.4.0 
  @backstage/catalog-model                         1.2.1 
  @backstage/cli-common                            0.1.12
  @backstage/cli                                   0.22.5
  @backstage/config-loader                         1.1.9 
  @backstage/config                                1.0.7 
  @backstage/core-app-api                          1.6.0 
  @backstage/core-components                       0.12.5
  @backstage/core-plugin-api                       1.5.0 
  @backstage/dev-utils                             1.0.13
  @backstage/errors                                1.1.5 
  @backstage/eslint-plugin                         0.1.2 
  @backstage/integration-aws-node                  0.1.2 
  @backstage/integration-react                     1.1.11
  @backstage/integration                           1.4.3 
  @backstage/plugin-api-docs                       0.9.1 
  @backstage/plugin-app-backend                    0.3.43
  @backstage/plugin-auth-backend                   0.18.1
  @backstage/plugin-auth-node                      0.2.12
  @backstage/plugin-catalog-backend-module-github  0.2.6 
  @backstage/plugin-catalog-backend                1.8.0 
  @backstage/plugin-catalog-common                 1.0.12
  @backstage/plugin-catalog-graph                  0.2.28
  @backstage/plugin-catalog-import                 0.9.6
  @backstage/plugin-catalog-node                   1.3.4
  @backstage/plugin-catalog-react                  1.4.0
  @backstage/plugin-catalog                        1.9.0
  @backstage/plugin-events-node                    0.2.4
  @backstage/plugin-github-actions                 0.5.16
  @backstage/plugin-home                           0.4.32
  @backstage/plugin-org                            0.6.6
  @backstage/plugin-permission-common              0.7.4
  @backstage/plugin-permission-node                0.7.6
  @backstage/plugin-permission-react               0.4.11
  @backstage/plugin-proxy-backend                  0.2.37
  @backstage/plugin-scaffolder-backend             1.12.0
  @backstage/plugin-scaffolder-common              1.2.6
  @backstage/plugin-scaffolder-node                0.1.1
  @backstage/plugin-scaffolder-react               1.2.0
  @backstage/plugin-scaffolder                     1.12.0
  @backstage/plugin-search-backend-module-pg       0.5.4
  @backstage/plugin-search-backend-node            1.1.4
  @backstage/plugin-search-backend                 1.2.4
  @backstage/plugin-search-common                  1.2.2
  @backstage/plugin-search-react                   1.5.1
  @backstage/plugin-search                         1.1.1
  @backstage/plugin-sentry                         0.5.1
  @backstage/plugin-stack-overflow-backend         0.1.12
  @backstage/plugin-stack-overflow                 0.1.12
  @backstage/plugin-tech-radar                     0.6.2
  @backstage/plugin-techdocs-backend               1.6.0
  @backstage/plugin-techdocs-module-addons-contrib 1.0.11
  @backstage/plugin-techdocs-node                  1.6.0
  @backstage/plugin-techdocs-react                 1.1.4
  @backstage/plugin-techdocs                       1.6.0
  @backstage/plugin-user-settings                  0.7.1
  @backstage/release-manifests                     0.0.9
  @backstage/test-utils                            1.2.6
  @backstage/theme                                 0.2.18
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by wedge-jarrad**
## Hey, I just made a Pull Request!

Documented functionality for providing files for download in techdocs. I spent quite a while trying to figure out how to to this and ended up finding the answer in Github issues submitted by other people looking for the same functionality. Thanks to @martinus for coming up with the answer.

Happy to move this somewhere else if someone can suggest a better place. 

Fixes #17734

Relates #13940 #6069

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by wedge-jarrad** [comment]
Not stale.

**Review by awanlin — APPROVED** [review]
This is great, thanks a ton @wedge-jarrad! 🚀 

Would you be interested in adding a working example of this to the Demo site here? https://github.com/backstage/demo/tree/master/docs/examples

For the failing build I'm not sure exactly how we can get around that, will need some suggestions from the Maintainers 🤔

**Comment by awanlin** [comment]
Hi @camilaibs @Rugvip, any suggestions how we can best resolve the CI failure? These are valid examples that are getting caught up in the link validation script: https://github.com/backstage/backstage/actions/runs/9393723111/job/25870264134?pr=25084#step:19:11

**Comment by wedge-jarrad** [comment]
> Would you be interested in adding a working example of this to the Demo site here? https://github.com/backstage/demo/tree/master/docs/examples

Yeah, I'd be happy to.

**Comment by awanlin** [comment]
Could use some help with how we can get the CI passing, this is a bit of an edge case 🤔

**Comment by freben** [comment]
Mmmh I would just remove the `md` from the triple ticks. Not worth it for the highlighting 🤔 Because it tries to really treat it as a proper subdocument with all of the rules that it entails.

Note that a rebase is needed as well.

**Comment by awanlin** [comment]
Still no luck on this one, any other ideas @freben?

File: docs/features/techdocs/how-to-guides.md [hunk 1]
```diff
@@ -874,3 +874,23 @@ metadata:
...
 TechDocs will notify the user that the page they are trying to access is no longer maintained. Then, they will be redirected. External site redirects are not supported. If an external redirect is provided, the user will instead be redirected to the index page of the documentation site.
+
+## Create download links for static assets
+
+You may want to make files available for download by your users such as PDF
+documents, images, or code templates. Download links for files included in your
+docs directory can be made by adding `{: download }` after a markdown link.
+
+```
+[Link text](./img/foo.jpg){: download }
```
Comment (freben) [inline docs/features/techdocs/how-to-guides.md:885]:
```suggestion
[Link text](https://example.com/foo.jpg){: download }
```

File: docs/features/techdocs/how-to-guides.md [hunk 2]
```diff
@@ -874,3 +874,23 @@ metadata:
...
+```
+
+The user's browser will download the file as `download.jpg` when the link is
+clicked.
+
+Specify a file name to control the name the file will be given when it is
+downloaded:
+
+```
+[Link text](./img/foo.jpg){: download="foo.jpg" }
```
Comment (freben) [inline docs/features/techdocs/how-to-guides.md:895]:
```suggestion
[Link text](https://example.com/foo.jpg){: download="foo.jpg" }
```

**Review by freben — APPROVED** [review]
Let's try to make them full URLs to circumvent the check

**Pull request merged**